presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -5866,6 +5866,25 @@ public void testUnionWithAggregation()
     }
 
     @Test
+    public void testUnionWithUnionAndAggregation()
+    {
+        assertQuery(
+                "SELECT count(*) FROM (" +
```

Comment:
Interestingly enough, the query being tested here works even without this patch.

On master, this doesn't work:

```sql
SELECT count(*) FROM (
    SELECT 1 FROM tpch.tiny.nation GROUP BY 1
    UNION ALL
    SELECT 1
    FROM (
      SELECT 1 FROM tpch.tiny.nation
      UNION ALL
      SELECT 1 FROM tpch.tiny.nation
    )
)
```

but this does:

```sql
SELECT count(*) FROM (
    SELECT 1 FROM tpch.tiny.nation GROUP BY 1
    UNION ALL
    (
      SELECT 1 FROM tpch.tiny.nation
      UNION ALL
      SELECT 1 FROM tpch.tiny.nation
    )
)
```